### DISCUSSION TITLE
How to use x-init in x-spread?

### DISCUSSION THREAD

**Discussion by maxsite** [alpinejs/alpine#904] [1. Help]
I want to rewrite such a code:

`<div x-data x-init="() => alert('x-init work')"></div>`

on such:
```
<!-- not work -->
<div x-data="myInit()" x-spread="trigger"></div>

<script>
function myInit() {
    return {
        trigger: {
            ['x-init']() {
                alert('x-init'); 
            },
        },
    }
}
</script>
```

Tell me, please, why doesn't it work?

**Comment by SimoTod** [comment]
As per doc, https://github.com/alpinejs/alpine#x-spread, you can't use x-init in x-spread.
The reason is that x-init happens before the component is fully initialized while x-spread works once the alpine reactivity kicks in.

**Comment by maxsite** [comment]
Thanks! In Russian documentation it is not written about this.
https://github.com/alpinejs/alpine/blob/master/README.ru.md#x-spread

**Reply by HugoDF** [reply]
Could you update those docs to have that note?

**Reply by maxsite** [reply]
I will try. :-)

**Comment by maxsite** [comment]
Instead of `x-init`, you can use `x-on:load.window`.

```
<div x-data="myInit()" x-spread="trigger"></div>

<script>
function myInit() {
    return {
        trigger: {
            ['x-on:load.window']() {
                alert('x-init'); 
            },
        },
    }
}
</script>
```

Perhaps this is useful to someone. :-)

**Reply by ryangjchandler** [reply]
@maxsite Be careful using the `load` event since this won't be fired until all images, resources, etc have been loaded so if you've got any super big images it will delay the initialisation.

**Reply by maxsite** [reply]
There is no other.: `document.onload` or `window.onload`. Usually there is enough.
